src/Core/ViewHelper/Traits/CompileWithContentArgumentAndRenderStatic.php

```diff
@@ -118,19 +127,30 @@ protected function buildRenderChildrenClosure()
      */
     public function resolveContentArgumentName()
     {
-        if (empty($this->contentArgumentName)) {
-            $registeredArguments = call_user_func_array([$this, 'prepareArguments'], []);
-            foreach ($registeredArguments as $registeredArgument) {
-                if (!$registeredArgument->isRequired()) {
-                    $this->contentArgumentName = $registeredArgument->getName();
-                    return $this->contentArgumentName;
-                }
+        // early return already defined argument name
+        if ($this->useContentArgumentName !== null) {
```

Comment:
Would a local `static` variable work, too? :thinking:

Reply:
mhhhh .....

Reply:
I thing static would introduce some nasty flaws, see: https://3v4l.org/r8C8D

Reply:
That's ... interesting. It surely makes sense somewhere but yes, too much risk here.